### PR TITLE
Added AWS STS related unit tests

### PR DESCRIPTION
**Pull request by ezio-auditore**
Signed-off-by: Kaustav Majumder <kmajumde@redhat.com>

### Explain the changes
1. Added missing AWS STS related unit tests

- [ ] Doc added/updated
- [x] Tests added

### CODE REVIEW

**Comment by ezio-auditore** [comment]
requesting review @guymguym @dannyzaken

File: src/test/unit_tests/test_aws_sts.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          };
+    const STSStub = sinon.stub(AWS, 'STS').callsFake(() => stsFake);
+
+    mocha.it('should generate aws sts creds', async function() {
+        const params = {
+            aws_sts_arn: roleArn
+        };
+        const expectedParams = [
+            { RoleArn: roleArn, RoleSessionName: 'testSession', WebIdentityToken: 'web-identity-token', DurationSeconds: defaultSTSCredsValidity }
+        ];
```
Comment (guymguym) [inline src/test/unit_tests/test_aws_sts.js:39]: we use the jsbeautifier settings - in vscode there's the HookyQR.beautify extension that will do it -
```suggest
        const expectedParams = [{
            RoleArn: roleArn,
            RoleSessionName: 'testSession',
            WebIdentityToken: 'web-identity-token',
            DurationSeconds: defaultSTSCredsValidity,
        }];
```

File: src/test/unit_tests/test_aws_sts.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        const additionalParams = {
+            RoleSessionName: 'testSession'
+        };
+        const s3 = await cloud_utils.createSTSS3Client(params, additionalParams);
+        dbg.log0('test.aws.sts.createSTSS3Client: ', s3);
+        assert.equal(s3.config.credentials.accessKeyId, fakeAccessKeyId);
+        assert.equal(s3.config.credentials.secretAccessKey, fakeSecretAccessKey);
+        assert.equal(s3.config.credentials.sessionToken, fakeSessionToken);
+        assert.equal(s3.config.region, 'us-east-1');
+        STSStub.restore();
```
Comment (guymguym) [inline src/test/unit_tests/test_aws_sts.js:63]: This restore() will not be called if this test fails before it.
The way to do it safer in mocha is to use mocha.before(...) and mocha.after(...)
See here for example:
https://github.com/ezio-auditore/noobaa-core/blob/92ec8ead6284c69badc581e9e9d1d5b7c2b624b6/src/test/unit_tests/test_namespace_fs.js#L53-L62

File: src/test/unit_tests/test_aws_sts.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+const mocha = require('mocha');
+const assert = require('assert');
+const sinon = require('sinon');
+const AWS = require('aws-sdk');
+const cloud_utils = require('../../util/cloud_utils');
+const dbg = require('../../util/debug_module')(__filename);
+const fs = require("fs");
+
+mocha.describe('AWS STS tests', function() {
```
Comment (guymguym) [inline src/test/unit_tests/test_aws_sts.js:12]: If you want this to run as part of the unit tests you should add it to `src/test/unit_tests/index.js`

**Review by guymguym — CHANGES_REQUESTED** [review]
Hi @ezio-auditore Thanks! 
I'm happy to see how creatively you used sinon, it really helps a lot with gray-box unit-testing.

Few questions I had:
- Is this meant to be testing just the cloud_utils code? 
- Is this testing the STS support we use as a client to our cloud backing stores or namespace stores? 
- Or is it for STS support of noobaa endpoints to its clients?
- In both cases I would define a test which is invoking this functionality as black-box like clients do (in case of backing stores it means to create a backing store with STS configuration and then invoke it while mocking the backing store).

Gray-box is great for really complex to sample logic where a tiny hook really slices it well, but is cloud_utils logic not too shallow for that? In any case, I'll let @dannyzaken review as well, and I don't mind having this as a base, and followup in subsequent PRs. Thanks!

**Comment by ezio-auditore** [comment]
> Hi @ezio-auditore Thanks! I'm happy to see how creatively you used sinon, it really helps a lot with gray-box unit-testing.
> 
> Few questions I had:
> 
> * Is this meant to be testing just the cloud_utils code?
> * Is this testing the STS support we use as a client to our cloud backing stores or namespace stores?
> * Or is it for STS support of noobaa endpoints to its clients?
> * In both cases I would define a test which is invoking this functionality as black-box like clients do (in case of backing stores it means to create a backing store with STS configuration and then invoke it while mocking the backing store).
> 
> Gray-box is great for really complex to sample logic where a tiny hook really slices it well, but is cloud_utils logic not too shallow for that? In any case, I'll let @dannyzaken review as well, and I don't mind having this as a base, and followup in subsequent PRs. Thanks!

This is meant to test just the cloud_utils code. Since I was not sure how I can mock the creation of backingstores.
Are there already existing tests I can refer to with regards to this?

**Review by dannyzaken — APPROVED** [review]
LGTM

**Comment by guymguym** [comment]
Hi @ezio-auditore, Thanks for the explanation, so I guess in that case I would just rename the file to `test_cloud_utils.js`.

**Review by guymguym — APPROVED** [review]

**Pull request merged**